src/API/Reports/Orders/Stats/DataStore.php

```diff
@@ -630,11 +630,14 @@ protected static function get_net_total( $order ) {
 	/**
 	 * Check to see if an order's customer has made previous orders or not
 	 *
-	 * @param array $order WC_Order object.
+	 * @param array     $order WC_Order object.
+	 * @param int|false $customer_id Customer ID. Optional.
 	 * @return bool
 	 */
-	public static function is_returning_customer( $order ) {
-		$customer_id = \Automattic\WooCommerce\Admin\API\Reports\Customers\DataStore::get_existing_customer_id_from_order( $order );
+	public static function is_returning_customer( $order, $customer_id = null ) {
+		if ( is_null( $customer_id ) ) {
```

Comment:
Out of curiosity, instead of passing the `$customer_id` I wonder if it be possible to directly check and cache the value of `get_report_customer_id()`.

Reply:
Sure, that could get added to the override classes. I'll try it out.

Reply:
I added some caching in 59c388366, but didn't eliminate the new `$customer_id` parameter here to avoid pushing (even more) knowledge of the override classes in to the data stores.

Reply:
Sorry for the very late reply on this, @jeffstieler!

In that case, I wonder if more of this `is_returning_customer` logic should be moved into the `Order` override class.  This would allow the caching you implemented in the order to continue working without passing around `$customer_id`.

Do you think that would work?

Reply:
Passing around the `$customer_id` here isn't ideal I agree, but I think moving all of the `is_returning_customer()` logic into the Order override class is problematic as well because of the calls to the Customers data store and `self::set_customer_first_order()`. Additionally, leaving it as-is allows the `is_returning_customer()` logic to work even if we aren't using the Order override class, which I think is a positive we'd like to maintain.

Reply:
Ya, that's a great point.  There is definitely a more ideal structure here, but this PR works well and is a big improvement over what's there right now.

I'll give this one last check so we can get this merged.